src/Amalgam/interpreter/InterpreterOpcodesMath.cpp

```diff
@@ -805,6 +805,60 @@ EvaluableNodeReference Interpreter::InterpretNode_ENT_MIN(EvaluableNode *en, boo
 	return EvaluableNodeReference::Null();
 }
 
+EvaluableNodeReference Interpreter::ConstructListWithZero()
+{
+	EvaluableNodeReference index_list;
+	index_list.SetReference(evaluableNodeManager->AllocNode(ENT_LIST));
+
+	index_list->GetOrderedChildNodesReference().push_back(AllocReturn(0.0, true)); 
+
+	return index_list;
+}
+
+EvaluableNodeReference Interpreter::InterpretNode_ENT_INDEX_MAX(EvaluableNode *en, bool immediate_result)
+{
+	auto &ocn = en->GetOrderedChildNodes();
+
+	if(ocn.size() == 0)
+		return EvaluableNodeReference::Null();
+
+	if(ocn.size() == 1) 
+	{
+		EvaluableNodeReference ocn_zero = InterpretNode(ocn[0]);
+
+		if(ocn_zero != nullptr && ocn_zero->GetType() == ENT_ASSOC)
+			return GetIndexMinMaxFromAssoc(ocn_zero, std::greater(), -std::numeric_limits<double>::infinity(), immediate_result);
+		else if(ocn_zero != nullptr && ocn_zero->GetType() == ENT_LIST)
+			return GetIndexMinMaxFromList(ocn_zero, ocn_zero->GetOrderedChildNodesReference(), std::greater(), -std::numeric_limits<double>::infinity(), immediate_result);
+		else
+			return ConstructListWithZero();
```

Comment:
So the reason I did it this way is because, in order to know the type of the first child node, I need to evaluate it (i.e. call InterpretNode on it). So if there is exactly one child node, and its not a list or assoc, the max index must be 0 (because its the only index).